gemd/entity/bounds/categorical_bounds.py

```diff
@@ -70,6 +70,53 @@ def contains(self, bounds: Union[BaseBounds, "BaseValue"]) -> bool:
 
         return bounds.categories.issubset(self.categories)
 
+    def union(self,
+              *others: Union["CategoricalBounds", "CategoricalValue"]
+              ) -> "CategoricalBounds":
+        """
+        Return the union of this bounds and other bounds.
+
+        The others list must also be Categorical Bounds or Values.
+
+        Parameters
+        ----------
+        others: Union[CategoricalBounds, CategoricalValue]
+            Other bounds or value objects to include.
+
+        Returns
+        -------
+        CategoricalBounds
+            The union of this bounds and the passed bounds
+
+        """
+        from gemd.entity.value.categorical_value import CategoricalValue
+
+        if any(not isinstance(x, (CategoricalBounds, CategoricalValue)) for x in others):
+            misses = {type(x).__name__
+                      for x in others
+                      if not isinstance(x, (CategoricalBounds, CategoricalValue))}
+            raise TypeError(f"union requires consistent typing; found {misses}")
+        result = self.categories.copy()
```

Comment:
Since this isn't a deep copy, won't the updates made below also affect this Bounds object?

Reply:
It shouldn't since I'm only copying the categories, not the bounds.  I think I'm verifying this with this test:
https://github.com/CitrineInformatics/gemd-python/blob/11861577b18c99eb473084ab46bd6a56d2714a38/gemd/entity/bounds/tests/test_categorical_bounds.py#L45-L47